sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientCache.scala

```diff
@@ -169,6 +168,19 @@ private[spark] object CosmosClientCache extends BasicLoggingTrait {
           builder.preferredRegions(cosmosClientConfiguration.preferredRegionsList.get.toList.asJava)
         }
 
+        if (cosmosClientConfiguration.enableClientTelemetry) {
+          System.setProperty(
+            "COSMOS.CLIENT_TELEMETRY_ENDPOINT",
+            cosmosClientConfiguration.clientTelemetryEndpoint.getOrElse(
+              "https://tools.cosmos.azure.com/api/clienttelemetry/trace"
+            ))
+          System.setProperty(
```

Comment:
we should only need one of these two? either set the property or enable in clientBuilder

Reply:
Not really - the builder sets the default value if the environment variable isn't present. Don't want to go through that debugging mess in Spark - set the config and be sure we fix the environment :-)